include/LieGroupControllers/impl/ProportionalDerivativeController/ControllerBase.h

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file ControllerBase.h
+ * @authors Giulio Romualdi
+ * @copyright This software may be modified and  distributed under the terms of
+ * the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef LIE_GROUP_CONTROLLERS_IMPL_PROPORTIONAL_DERIVATIVE_CONTROLLER_CONTROLLER_BASE_H
+#define LIE_GROUP_CONTROLLERS_IMPL_PROPORTIONAL_DERIVATIVE_CONTROLLER_CONTROLLER_BASE_H
+
+#include <LieGroupControllers/impl/ControllerBase.h>
+#include <tuple>
+
+namespace LieGroupControllers
+{
+template <typename _Derived>
+class ProportionalDerivativeControllerBase : public ControllerBase<_Derived>
+{
+    using State = typename ControllerBase<_Derived>::State;
+    using Vector = typename ControllerBase<_Derived>::Vector;
+    using Gains = typename ControllerBase<_Derived>::Gains;
+
+    State m_state;
+    State m_desiredState;
+    Vector m_feedForward;
+    Vector m_controlOutput;
+    Gains m_gain;
+
+public:
+    bool setState(const State& state);
+
+    bool setDesiredState(const State& state);
+
+    bool setFeedForward(const Vector& feedForward);
+
+    void computeControlLaw();
+
+    void setGain(const Gains& gain);
+
+    const Vector& getControl() const;
```

Comment:
What happens if one of these set methods is not actually used? Is manif taking care of proper initializations under the hood?

Reply:
Note that `manif` does not do any initialization at construction.

Reply:
If compiled in debug. the following error is thrown by `manif` 
```
SO3 constructor argument not normalized !
```

Reply:
Because the input quaternion is not 1, 0,0,0.